great_expectations/expectations/metrics/column_aggregate_metrics/column_distinct_values.py

```diff
@@ -93,21 +123,23 @@ def _sqlalchemy(
         metric_value_kwargs: Dict,
         metrics: Dict[str, Any],
         runtime_configuration: Dict,
-    ):
-        observed_value_counts = metrics["column.value_counts"]
-        return len(observed_value_counts)
+    ) -> bool:
+        threshold = metric_value_kwargs.get("threshold")
+        column_distinct_values_count = metrics.get("column.distinct_values.count")
+        return column_distinct_values_count < threshold
 
     @metric_value(engine=SparkDFExecutionEngine)
     def _spark(
         cls,
-        execution_engine: SqlAlchemyExecutionEngine,
+        execution_engine: SparkDFExecutionEngine,
```

Comment:
This seems like an odd parameter since this is called `_spark` and it has a spark decorator. We don't use it here but I'm guessing some interface requires it? It seems like it should have a default or be factored out (in future work). Does that seem reasonable/desirable?

Reply:
That makes sense to me. For now I have removed this and replaced it (and other unused parameters with **kwargs).